src/librustc_mir/interpret/eval_context.rs

```diff
@@ -291,22 +291,41 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         ty.is_freeze(*self.tcx, self.param_env, DUMMY_SP)
     }
 
-    pub(super) fn subst_and_normalize_erasing_regions<T: TypeFoldable<'tcx>>(
+    /// Call this whenever you have a value that `needs_subst`. Not guaranteed to actually
+    /// monomorphize the value. If we are e.g. const propagating inside a generic function, some
+    /// things may depend on a generic parameter and thus can never be monomorphized.
+    pub(super) fn subst_and_normalize_erasing_regions_in_frame<T: TypeFoldable<'tcx>>(
```

Comment:
@RalfJung came up with `subst_from_frame_and_normalize_erasing_regions`, I kinda like that more?

Reply:
In particular this finally made it click for me in terms of: we need this when taking something from the current frame to move it into the `InterpCx` "universe".

Also see the comments I wrote for my own experiments [here](https://github.com/RalfJung/rust/commit/aa03becdaf22af84f816ab861157e6c57f4572c1#diff-470c35c1f39f9d3f8817435a63c73ad2R317) and [here](https://github.com/RalfJung/rust/commit/aa03becdaf22af84f816ab861157e6c57f4572c1#diff-470c35c1f39f9d3f8817435a63c73ad2R337); I think it would be useful to carry those over.